### PR TITLE
fix(account-lib): STLX-1902 Fixed toBroadcastFormat method

### PR DESCRIPTION
**Pull request by smaria95**
Manually set owners and target arguments from deploy to TX JSON before return stringified transaction.

Jira Ticket: [STLX-1902](https://bitgoinc.atlassian.net/browse/STLX-1902)

### CODE REVIEW

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 1]
```diff
@@ -102,13 +105,38 @@ export class Transaction extends BaseTransaction {
...
-   * if there are outputs. For transactions without outputs (e.g. wallet initializations),
-   * this function will not do anything
-   */
-  loadInputsAndOutputs(): void {
-    throw new NotImplementedError('loadInputsAndOutputs not implemented');
+  setOwnersInJson(txJson: Record<string, any>): void {
+    if (this.casperTx.session.isModuleBytes()) {
+      const ownersValues = new Map();
+
+      [0, 1, 2].forEach(index => {
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:112]: create consts for magic numbers

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 2]
```diff
@@ -102,13 +105,38 @@ export class Transaction extends BaseTransaction {
...
+        ownersValues.set(
+          OWNER_PREFIX + index,
+          (this.casperTx.session.getArgByName(OWNER_PREFIX + index) as CLValue).asString(),
+        );
+      });
+
+      txJson['deploy']!['session']['ModuleBytes']['args'].forEach(arg => {
+        if (ownersValues.has(arg[0])) {
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:120]: create consts for magic numbers

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 3]
```diff
@@ -102,13 +105,38 @@ export class Transaction extends BaseTransaction {
...
+          arg[1]['parsed'] = ownersValues.get(arg[0]);
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:121]: create consts for magic numbers

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 4]
```diff
@@ -102,13 +105,38 @@ export class Transaction extends BaseTransaction {
...
+
+  setTransfersFieldsInJson(txJson: Record<string, any>): void {
+    if (this.casperTx.session.isTransfer()) {
+      const transferValues = new Map();
+      transferValues.set('amount', getTransferAmount(this.casperTx.session));
+      transferValues.set('to_address', getTransferDestinationAddress(this.casperTx.session));
+      transferValues.set('id', getTransferId(this.casperTx.session).toString());
+
+      txJson['deploy']!['session']['Transfer']['args'].forEach(arg => {
+        if (transferValues.has(arg[0])) {
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:135]: create consts for magic numbers

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 5]
```diff
@@ -102,13 +105,38 @@ export class Transaction extends BaseTransaction {
...
+          arg[1]['parsed'] = transferValues.get(arg[0]);
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:136]: create consts for magic numbers

File: modules/account-lib/test/unit/coin/cspr/transaction.ts [hunk 6]
```diff
@@ -90,6 +114,49 @@ describe('Cspr Transaction', () => {
   });
 
   describe('should return encoded tx', function() {
+    it('wallet initialization', async function() {
+      const walletInitTx = await getWalletInitTransaction();
+      const encodedTx = walletInitTx.toBroadcastFormat();
+      const walletInitJsonTx = JSON.parse(encodedTx);
+      const ownersValues = new Map();
+
+      [0, 1, 2].forEach(index => {
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transaction.ts:123]: create consts for magic numbers

File: modules/account-lib/test/unit/coin/cspr/transaction.ts [hunk 7]
```diff
@@ -90,6 +114,49 @@ describe('Cspr Transaction', () => {
...
+        ownersValues.set(
+          OWNER_PREFIX + index,
+          (walletInitTx.casperTx.session.getArgByName(OWNER_PREFIX + index) as CLValue).asString(),
+        );
+      });
+
+      const jsonOwnerArgs = walletInitJsonTx['deploy']['session']['ModuleBytes']['args'].filter(arg =>
+        ownersValues.has(arg[0]),
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transaction.ts:131]: create consts for magic numbers

File: modules/account-lib/test/unit/coin/cspr/transaction.ts [hunk 8]
```diff
@@ -90,6 +114,49 @@ describe('Cspr Transaction', () => {
...
+      );
+      jsonOwnerArgs.length.should.equal(ownersValues.size);
+
+      jsonOwnerArgs.forEach(arg => {
+        arg[1]['parsed'].should.be.equal(ownersValues.get(arg[0]));
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transaction.ts:136]: create consts for magic numbers

File: modules/account-lib/test/unit/coin/cspr/transaction.ts [hunk 9]
```diff
@@ -90,6 +114,49 @@ describe('Cspr Transaction', () => {
...
+      const encodedTx = transferTx.toBroadcastFormat();
+      const transferJsonTx = JSON.parse(encodedTx);
+      const transferValues = new Map();
+
+      transferValues.set('amount', getTransferAmount(transferTx.casperTx.session));
+      transferValues.set('to_address', getTransferDestinationAddress(transferTx.casperTx.session));
+      transferValues.set('id', getTransferId(transferTx.casperTx.session).toString());
+
+      const jsonOwnerArgs = transferJsonTx['deploy']['session']['Transfer']['args'].filter(arg =>
+        transferValues.has(arg[0]),
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transaction.ts:151]: create consts for magic numbers

File: modules/account-lib/test/unit/coin/cspr/transaction.ts [hunk 10]
```diff
@@ -90,6 +114,49 @@ describe('Cspr Transaction', () => {
...
+      );
+      jsonOwnerArgs.length.should.equal(transferValues.size);
+
+      jsonOwnerArgs.forEach(arg => {
+        arg[1]['parsed'].should.be.equal(transferValues.get(arg[0]));
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transaction.ts:156]: create consts for magic numbers

**Review by rliberal-simtlix — CHANGES_REQUESTED** [review]

**Review by rliberal-simtlix — APPROVED** [review]
LGTM

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**